src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -13,6 +13,9 @@ public class StringFunctions {
     private static final char SPACE = ' ';
 
     public static final String reverseWithStringConcat(String string) {
+        if (string == null || string.isEmpty()) {
+            return "";  // Return an empty string if input is null or empty
```

Comment:
Ensured the reverseWithStringBuilder method also handles null or empty input, improving the robustness of the method.